pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -252,17 +254,25 @@ func (a *AgentPXEFiles) handleAdditionals390xArtifacts(bootArtifactsFullPath str
 		return err
 	}
 
-	agentINSFile := filepath.Join(bootArtifactsFullPath, fmt.Sprintf("%s.%s-generic.ins", a.filePrefix, a.cpuArch))
-	genericReader, err := os.Open(filepath.Join(a.tmpPath, "generic.ins"))
+	// Read and update generic.ins
+	genericInsPath := filepath.Join(a.tmpPath, "generic.ins")
+	data, err := os.ReadFile(genericInsPath)
 	if err != nil {
 		return err
 	}
-	defer genericReader.Close()
 
-	err = copyfile(agentINSFile, genericReader)
+	// Replace file names with dynamically generated names
+	updatedContent := strings.ReplaceAll(string(data), "kernel.img", fmt.Sprintf("%s.%s-vmlinuz", a.filePrefix, a.cpuArch))
+	updatedContent = strings.ReplaceAll(updatedContent, "initrd.img", fmt.Sprintf("%s.%s-initrd.img", a.filePrefix, a.cpuArch))
+	updatedContent = strings.ReplaceAll(updatedContent, "initrd.addrsize", fmt.Sprintf("%s.%s-initrd.addrsize", a.filePrefix, a.cpuArch))
+
+	// Write the updated content back
+	agentINSFile := filepath.Join(bootArtifactsFullPath, fmt.Sprintf("%s.%s-generic.ins", a.filePrefix, a.cpuArch))
+	err = os.WriteFile(agentINSFile, []byte(updatedContent), 0644)
 	if err != nil {
 		return err
 	}
 
+	logrus.Infof("Updated generic.ins with correct filenames: %s", agentINSFile)
```

Comment:
I don't think this log line is needed and maybe confusing. If needed, perhaps it should be Debug instead of Info.
```suggestion
```